### ISSUE TITLE
Batching leads to different tensors

### ISSUE THREAD

**Issue by adrianeboyd** [explosion/spacy-transformers#132]
Batching leads to different tensors with transformers models on both CPU and GPU.

Previously discussed here: https://github.com/explosion/spaCy/issues/4766

Here's an example script to reproduce this issue (from https://github.com/explosion/spaCy/issues/4766#issuecomment-577870808):

```
import spacy
import numpy as np

nlp = spacy.load('en_trf_bertbaseuncased_lg')

messages = ["Hello my name is daniel", "I would like to participate in interesting events",
            "something is very strange with SpaCy", "The vectors aren't similar",
            "let us try to round both vectors to 5 decimals"]

docs_by_batch = nlp.pipe(messages, batch_size=2)
docs_standard = [nlp(message) for message in messages]

for doc_by_batch, doc_standard in zip(docs_by_batch, docs_standard):
    if np.array_equal(np.around(doc_by_batch.tensor, 2), np.around(doc_standard.tensor, 2)):
        print("Identical Vectors")
    else:
        print("Different Vectors")
```

In some quick tests, I saw differences with all `spacy-transformers` models and I couldn't see an immediate pattern in terms of batch size or the position of a document within a batch.

**Comment by ZhuoruLin** [comment]
Hi @adrianeboyd, I think we might bump into the same issue with `trf_tok2vec`. I would be grateful if you can take a look at issue #135 and tell me what you think.

**Comment by adrianeboyd** [comment]
I think this should be fixed by #136.

See the script in #136 and note that `numpy.allclose()` is more suitable than `numpy.around()` for testing.
